tests/pruning/test_model_pruning_analysis.py

```diff
@@ -0,0 +1,70 @@
+"""
+ Copyright (c) 2020 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import pytest
+
+from nncf.dynamic_graph.context import Scope
+from tests.helpers import create_compressed_model_and_algo_for_test
+from tests.pruning.helpers import PruningTestModelEltwise, get_basic_pruning_config, TestModelBranching
+
+
+@pytest.mark.parametrize("model,not_pruned_modules,pruned_groups,prune_first,prune_last,prune_downsample",
```

Comment:
Please consider making your test case definitions more explicit by using a dedicated struct with descriptive fields - review `tests\test_input_management.py` for an example of such struct usage.